ndsl/__init__.py

```diff
@@ -1 +1,31 @@
+from .checkpointer import SnapshotCheckpointer
+from .comm import (
+    CachingCommReader,
+    CachingCommWriter,
+    ConcurrencyError,
+    CubedSphereCommunicator,
+    CubedSpherePartitioner,
+    LocalComm,
+    NullComm,
+    TileCommunicator,
+    TilePartitioner,
+)
```

Comment:
It looks a little strange to expose `CachingComm`, `LocalComm`, and `NullComm` but not `MPIComm`, when `MPIComm` is the most-used communicator. I'm ok with exposing all of the comms, or only doing `MPIComm`, or maybe keeping all of them inside `ndsl.comm` but exposing all of them except `MPIComm` doesn't seem great to me. There's also a question of whether to expose the `Communicator` base class for typing purposes